libs/utils/src/cardTallies.ts

```diff
@@ -416,3 +423,135 @@ export const compressTally = (
     throwIllegalValue(contest)
   })
 }
+
+const getContestTalliesForSerializedContest = (
+  contest: AnyContest,
+  serializedContest:
+    | SerializedCandidateVoteTally
+    | SerializedYesNoVoteTally
+    | SerializedMsEitherNeitherTally
+): ContestTally[] => {
+  switch (contest.type) {
+    case 'yesno': {
+      const yesNoTally = serializedContest as SerializedYesNoVoteTally
+      return [
+        {
+          contest,
+          tallies: {
+            yes: { option: ['yes'], tally: yesNoTally.yes },
+            no: { option: ['no'], tally: yesNoTally.no },
+          },
+          metadata: {
+            undervotes: yesNoTally.undervotes,
+            overvotes: yesNoTally.overvotes,
+            ballots: yesNoTally.ballotsCast,
+          },
+        },
+      ]
+    }
+    case 'candidate': {
+      const candidateTally = serializedContest as SerializedCandidateVoteTally
+      const candidateTallies: Dictionary<ContestOptionTally> = {}
+      let index = 0
+      contest.candidates.forEach((candidate) => {
+        candidateTallies[candidate.id] = {
+          option: candidate,
+          tally: candidateTally.candidates[index] ?? 0,
+        }
+        index += 1
+      })
```

Comment:
safer / cleaner to do

`forEach((candidate, candidateIndex) => {`

then use `candidateIndex` instead of manually incrementing your own index.